status/manager.go

```diff
@@ -80,26 +134,34 @@ func (s *Manager) SubServerStatus(_ context.Context,
 
 // RegisterSubServer will create a new sub-server entry for the Manager to
 // keep track of.
-func (s *Manager) RegisterSubServer(name string) {
+func (s *Manager) RegisterSubServer(name string, opts ...SubServerOption) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	ss := newSubServerStatus()
-	ss.manualStatus = "Registered"
-
-	s.subServers[name] = ss
+	s.registerSubServerUnsafe(name, true, opts...)
 }
 
 // RegisterAndEnableSubServer will create a new sub-server entry for the
 // Manager to keep track of and will set it as enabled.
-func (s *Manager) RegisterAndEnableSubServer(name string) {
+func (s *Manager) RegisterAndEnableSubServer(name string,
+	opts ...SubServerOption) {
+
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
-	ss := newSubServerStatus()
-	ss.Disabled = false
+	s.registerSubServerUnsafe(name, false, opts...)
+}
+
+func (s *Manager) registerSubServerUnsafe(name string, disabled bool,
+	opts ...SubServerOption) {
+
+	ss := newSubServerStatus(disabled)
```

Comment:
nit: just for future, this refactor along with the renames later on could have been done in 2 commits at the start of the PR. That way, all the commits that follow are mainly additions. 

but yeah - just heads up for future PRs :) no need to change here